### DISCUSSION TITLE
Start/Stop Confirmation

### DISCUSSION THREAD

**Discussion by bditter** [BigThunderSR/homeassistant-addons-onstar2mqtt#560] [Q&A] [answered]
Is there a way that a sensor could be provided to accurately depict whether the engine is indeed running or stopped during a remote start/stop. I was hoping that I could look at the RPMs after sending remote start/stop, but the RPM command fails and the Engine RPM consistently reads zero.

**Comment by BigThunderSR** [comment]
We can only read the status that the OnStar API gives us which unfortunately isn't correct (as you state) when the OnStar module in the vehicle has gone into power saving mode after the vehicle has been turned off for a while. If the start or stop command status monitor sensor comes back with a status of "Completed Successfully", it is very likely that the engine is either running or stopped for each respective command. However, there is no way we can guarantee status, so it's the responsibility of the user to confirm the real status of their vehicle.

I personally use my node-RED module (https://github.com/BigThunderSR/node-red-contrib-onstar2) within the HA node-RED add-on to start my vehicles because I can confirm with better accuracy that the vehicle did start based on the message that comes back along with the timestamp the vehicle started at. Because of how HA expects auto-created MQTT sensor configs to be defined, it's challenging to parse these in the onstar2mqtt code in the current form without some significant refactoring and I don't know when I'll have time to do that in the near future, so my node-RED module is your best option for now.

**Reply by bditter** [reply]
Thank you for the information. I just installed node-red ass-on to my home assistant. I've never used node-red so if you have a sample flow that I could see/use to accomplish my goal it would be greatly appreciated. 

I want to produce a sensor inside home assistant getting information from node-red that the vehicle is running or not. Obviously this is not guaranteed based on the information you provided but I'd assume it would be accurate most of the time.

As a side note: The RPM never returns anything other than zero even if the car is running. Is the RPM only for EVs? Just for information my vehicle is a 2022 Chevrolet Camaro.

**Reply by BigThunderSR** [reply]
The issue is that the vehicle doesn't send back any diagnostic data until you get in and start driving, so not even the RPM will come back until then. Both my vehicles are ICE and it's not an EV related limitation.

I'll have to get back to you on the other ask.

**Reply by BigThunderSR** [reply]
Sorry for the delay in a response. I've been thinking of a couple of ways to get you something going, but it will take me a bit longer due to some personal priorities I need to focus on right now.

**Reply by BigThunderSR** [reply, marked as answer]
Command success/fail status is reported via auto-generated command status sensors starting at v1.15.0.

**Comment by BigThunderSR** [comment]
Command success/fail status is reported via auto-generated command status sensors starting at v1.15.0.
